controllers/datasciencecluster/datasciencecluster_controller_support.go

```diff
@@ -0,0 +1,191 @@
+	instance *dscv1.DataScienceCluster,
+	component cr.ComponentHandler,
+) error {
+	ms := component.GetManagementState(instance)
+	ci := component.NewCRObject(instance)
+
+	switch ms {
+	case operatorv1.Managed:
+		err := ctrl.SetControllerReference(instance, ci, cli.Scheme())
+		if err != nil {
+			return err
+		}
+		err = cli.Apply(ctx, ci, client.FieldOwner(fieldOwner), client.ForceOwnership)
+		if err != nil {
+			return client.IgnoreNotFound(err)
+		}
+	case operatorv1.Removed:
+		err := cli.Delete(ctx, ci, client.PropagationPolicy(metav1.DeletePropagationForeground))
+		if err != nil {
+			return client.IgnoreNotFound(err)
+		}
+	default:
+		return fmt.Errorf("unsupported management state: %s", ms)
+	}
+
+	return nil
+}
+
+// setAvailability updates the "Available" condition of the DataScienceCluster instance  based on the provided error.
+// If the error is nil, the condition is set to True with a success message.
+// If an error is present, the condition is set to False with the error message.
+//
+// Parameters:
+//   - instance: The DataScienceCluster instance whose status conditions are being updated.
+//   - err: The error encountered during reconciliation, if any.
+//
+// Returns:
+//   - A boolean indicating whether the "Available" condition is True
+func setAvailability(instance *dscv1.DataScienceCluster, err error) bool {
+	condition := conditionsv1.Condition{
+		Type:    conditionsv1.ConditionAvailable,
+		Status:  corev1.ConditionTrue,
+		Reason:  status.AvailableReason,
+		Message: "DataScienceCluster resource reconciled successfully",
+	}
+
+	if err != nil {
+		condition.Status = corev1.ConditionFalse
+		condition.Reason = status.DegradedReason
+		condition.Message = fmt.Sprintf("DataScienceCluster resource reconciled with errors: %v", err)
+	}
+
+	conditionsv1.SetStatusCondition(&instance.Status.Conditions, condition)
+
+	return condition.Status == corev1.ConditionTrue
+}
+
+// setReadiness updates the "Ready" condition and the phase of the DataScienceCluster instance based on the list of
+// non-ready components.
+//
+// If all components are ready, the condition is set to True  with a success message.
+// If any components are not ready, the condition is set to False  with the error message.
+//
+// Parameters:
+//   - instance: The DataScienceCluster instance whose status conditions are being updated.
+//   - nonReadyComponents: A slice of strings representing the names of components that are not ready.
+//
+// Returns:
+//   - A boolean indicating whether the "Ready" condition is True.
+func setReadiness(instance *dscv1.DataScienceCluster, nonReadyComponents []string) bool {
+	condition := conditionsv1.Condition{
+		Type:    conditionsv1.ConditionType(status.ConditionTypeReady),
+		Status:  corev1.ConditionTrue,
+		Reason:  status.ReadyReason,
+		Message: status.ReadyReason,
+	}
+
+	slices.Sort(nonReadyComponents)
+
+	if len(nonReadyComponents) != 0 {
+		instance.Status.Phase = status.PhaseNotReady
```

Comment:
Probably does not make sense since overwritten on line 87?

Reply:
fixed